### PR TITLE
skip proxy udp port

### PR DESCRIPTION
**Pull request by ChenLingPeng**
Signed-off-by: forrestchen <forrestchen@tencent.com>

Please provide a description for what this PR is for.
skip udp port redirect

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by ChenLingPeng** [comment]
@howardjohn test added.
It seems the failed tests not related to this PR.

**Comment by ayj** [comment]
To echo @howardjohn's, the MCP changes seem unrelated to this PR. They do look correct but I'd prefer to see them in a separate PR. In general it difficult to review/bisect/revert PR when they include multiple unrelated changes.

File: pkg/kube/inject/inject.go [hunk 1]
```diff
@@ -843,6 +843,9 @@ func intoObject(sidecarTemplate string, valuesConfig string, meshconfig *meshcon
 func getPortsForContainer(container corev1.Container) []string {
 	parts := make([]string, 0)
 	for _, p := range container.Ports {
+		if p.Protocol == corev1.ProtocolUDP {
```
Comment (hzxuzhonghu) [inline pkg/kube/inject/inject.go]: Should also filter out SCTP

Comment (ChenLingPeng) [reply pkg/kube/inject/inject.go]: fixed

**Comment by ChenLingPeng** [comment]
@ayj @howardjohn the MCP changes have been removed

**Pull request merged**